multi-educator/src/Dscp/MultiEducator/Launcher/Mode.hs

```diff
@@ -111,35 +111,36 @@ lookupEducator login = do
     mctx <- ask
     let MultiEducatorResources{ _merEducatorData }
             = mctx ^. lensOf @MultiEducatorResources
-    fmap (\(EducatorContexts ctxs) -> M.lookup login ctxs) . atomically $ readTVar _merEducatorData
-
-loadEducator :: (MultiEducatorWorkMode ctx m) => Bool -> Text -> Maybe PassPhrase -> m Bool
-loadEducator _new login mpassphrase = do
+    EducatorContexts ctxs <- atomically $ readTVar _merEducatorData
+    case M.lookup login ctxs of
+        Just res -> return $ Just res
+        Nothing -> Just <$> loadEducator login Nothing
```

Comment:
It looks suspicious that we never write to `_merEducatorData`. Could you fix this now as well?

Reply:
Done, I forgot about this.

Anyway this couple of functions (especially `loadEducator`) should be modified, but I don't know how much it is necessary for this issue.
There are a couple of changes that could probably be done already (e.g. `lookupEducator` does not have to return a `Maybe` anymore), but I have not done those yet because I'm firstly trying to understand what exactly I should do here.

Reply:
@Martoon-00 I just remembered that this was not in `lookupEducator` because it is already handled at the end of `loadEducator`.

I suppose I could remove the `lookup` check here and revert the last commit.

Reply:
Oh lol, it's using `lensOf` to access contexts `TVar`, my bad.

Reply:
No problem, I searched `_merEducatorData` as well and though I forgot about the update, it's probably better if I make this visible as well.